pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -78,12 +78,14 @@ type cloudCIDRAllocator struct {
 
 	recorder record.EventRecorder
 	queue    workqueue.RateLimitingInterface
+
+	allocatorParams CIDRAllocatorParams
```

Comment:
we may able to derive the information here from `CIDRAllocatorParams` and create our internal enum or another format to simpligy the logic inside the controller

Reply:
based on the ServiceCIDRs, this can be IPv4, IPv6, IPv4IPv6 or IPv6IPv4

Reply:
I added an enum to address this comment.